miniscope_io/stream_daq.py

```diff
@@ -376,67 +416,39 @@ def _format_frame(
         """
         locallogs = init_logger("streamDaq.frame")
-        header_data = None
+        try:
+            for frame_data in exact_iter(frame_buffer_queue.get, None):
+                if self.terminate.is_set():
+                    break
 
-        while 1:
-            if frame_buffer_queue.qsize() > 0:  # Higher is safe but lower is fast.
                 locallogs.debug("Found frame in queue")
-
-                frame_data = frame_buffer_queue.get()  # pixel data for single frame
-                nbit_lost = 0
-
-                for i, npix_expected in enumerate(self.buffer_npix):
-                    if frame_data[i] is not None:
-                        header_data, fm_dat = self._parse_header(
-                            Bits(frame_data[i]), truncate="header"
-                        )
-                    else:
-                        frame_data[i] = Bits(int=0, length=npix_expected * self.config.pix_depth)
-                        nbit_lost += npix_expected
-                        continue
-                    npix_header = header_data.pixel_count
-                    npix_actual = len(fm_dat) / self.config.pix_depth
-
-                    if npix_actual != npix_expected:
-                        if i < len(self.buffer_npix) - 1:
-                            locallogs.warning(
-                                f"Pixel count inconsistent for frame {header_data.frame_num} "
-                                f"buffer {header_data.frame_buffer_count}. "
-                                f"Expected: {npix_expected}, "
-                                f"Header: {npix_header}, "
-                                f"Actual: {npix_actual}"
-                            )
-                        nbit_expected = npix_expected * self.config.pix_depth
-                        if len(fm_dat) > nbit_expected:
-                            fm_dat = fm_dat[:nbit_expected]
-                        else:
-                            nbit_pad = nbit_expected - len(fm_dat)
-                            fm_dat = fm_dat + Bits(int=0, length=nbit_pad)
-                            nbit_lost += nbit_pad
-
-                    frame_data[i] = fm_dat
-
-                pixel_vector = frame_data[0]
-                for d in frame_data[1:]:
-                    pixel_vector = pixel_vector + d
-
-                assert len(pixel_vector) == (
-                    self.config.frame_height * self.config.frame_width * self.config.pix_depth
-                )
+                if len(frame_data) == 0:
+                    continue
+                frame_data = np.concatenate(frame_data, axis=0)
 
                 if self.config.LSB:
-                    pixel_vector = Array(
-                        "uint:32",
-                        [
-                            pixel_vector[i : i + 32][::-1].uint
-                            for i in reversed(range(0, len(pixel_vector), 32))
-                        ],
+                    frame_data = np.flip(frame_data)
```

Comment:
This flips the pixel order within the entire image, and the `BufferFormatter` one flips within each 4-byte word. So I assume this will flip the image by 180 deg.

I don't think this is necessary, but I kept it because I was too lazy to change the hashes for the test data.